crates/biome_js_analyze/src/lint/nursery/no_magic_numbers.rs

```diff
@@ -0,0 +1,240 @@
+    // Ignore specific values from config
+    if config.ignore.contains(&value) {
+        return true;
+    }
+
+    // Check parent nodes for various allowed contexts
+    if let Some(parent) = node.parent() {
+        match parent.kind() {
+            // Array index check
+            // k if k == JsSyntaxKind::JS_STATIC_MEMBER_EXPRESSION
+            //     || k == JsSyntaxKind::JS_COMPUTED_MEMBER_EXPRESSION =>
+            // {
+            //     if config.ignore_array_indexes
+            //         && JsMemberExpression::cast(parent)
+            //             .map_or(false, |m| m.object().is_ok() && is_valid_array_index(value))
+            //     {
+            //         return true;
+            //     }
+            // }
+            // parseInt radix check
+            k if k == JsSyntaxKind::JS_CALL_EXPRESSION => {
+                if let Some(call) = JsCallExpression::cast(parent) {
+                    if let Ok(callee) = call.callee() {
+                        if callee.syntax().text() == "parseInt" {
+                            return true;
+                        }
+                    }
+                }
+            }
+            // Default value check
+            k if k == JsSyntaxKind::JS_OBJECT_ASSIGNMENT_PATTERN
+                || k == JsSyntaxKind::JS_ARRAY_ASSIGNMENT_PATTERN =>
+            {
+                if config.ignore_default_values {
+                    return true;
+                }
+            }
+            // Class field check
+            k if k == JsSyntaxKind::JS_PROPERTY_CLASS_MEMBER => {
+                if config.ignore_class_field_initial_values {
+                    return true;
+                }
+            }
+            _ => {}
+        }
+    }
+
+    if !config.detect_objects {
+        if let Some(parent) = node.parent() {
+            match parent.kind() {
+                JsSyntaxKind::JS_OBJECT_EXPRESSION | JsSyntaxKind::JS_PROPERTY_OBJECT_MEMBER => {
+                    return true;
+                }
+                JsSyntaxKind::JS_ASSIGNMENT_EXPRESSION => {
+                    if let Some(expr) = JsAssignmentExpression::cast(parent) {
+                        if let Ok(left_pattern) = expr.left() {
+                            if left_pattern.as_js_object_assignment_pattern().is_none() {
+                                return true;
+                            }
+                        }
+                    }
+                }
+                _ => {}
+            }
+        }
+    }
+
+    false
+}
+
+fn check_const_violation(
+    node: &SyntaxNode<biome_js_syntax::JsLanguage>,
+    config: &NoMagicNumbersConfig,
+) -> bool {
+    if !config.enforce_const {
+        return false;
+    }
+
+    node.ancestors()
+        .find_map(JsVariableDeclarator::cast)
+        .and_then(|decl| decl.syntax().parent())
+        .and_then(JsVariableDeclarator::cast)
+        .and_then(|parent| parent.id().ok())
+        .map_or(false, |kind| kind.to_string() != "const")
```

Comment:
Don't convert to `String` when you just need a `&str`. Look for `text_trimmed` method